src/SFA.DAS.EmployerIncentives/Application/Commands/UpdateVendorRegistrationFormCaseStatus/RefreshVendorRegistrationFormCaseStatusCommandHandler.cs

```diff
@@ -48,25 +52,31 @@ public async Task<DateTime> Handle(RefreshVendorRegistrationFormCaseStatusComman
                 _logger.LogError($"[VRF Refresh] [SkipCode={response.SkipCode}] returned by the Finance API with parameters: [DateTimeFrom={request.FromDateTime.ToIsoDateTime()}] [DateTimeTo={request.ToDateTime.ToIsoDateTime()}]", request.FromDateTime, request.ToDateTime);
             }
 
-            FindLatestUpdateForEachLegalEntity(response);
-
-            Task UpdateVendorRegistrationCaseStatus(VendorRegistrationCase @case)
-            {
-                return _incentivesService.UpdateVendorRegistrationCaseStatus(
-                    new UpdateVendorRegistrationCaseStatusRequest
-                    {
-                        CaseId = @case.CaseId,
-                        HashedLegalEntityId = @case.ApprenticeshipLegalEntityId,
-                        Status = @case.CaseStatus,
-                        CaseStatusLastUpdatedDate = @case.CaseStatusLastUpdatedDate
-                    });
-            }
+            FindLatestUpdateForEachLegalEntity(response);                      
 
             await Task.WhenAll(response.RegistrationCases.Select(UpdateVendorRegistrationCaseStatus));
 
             return await Task.FromResult(nextRunDateTime);
         }
 
+        private async Task UpdateVendorRegistrationCaseStatus(VendorRegistrationCase @case)
+        {
+            var accountLegalEntity = await _incentivesService.GetLegalEntityByHashedId(@case.ApprenticeshipLegalEntityId);
+            if (!VendorIdIsPopulated(accountLegalEntity?.VrfVendorId))
+            {
+                await _mediator.Send(new GetAndAddEmployerVendorIdCommand(@case.ApprenticeshipLegalEntityId));
+            }
+
+            await _incentivesService.UpdateVendorRegistrationCaseStatus(
+                new UpdateVendorRegistrationCaseStatusRequest
+                {
+                    CaseId = @case.CaseId,
+                    HashedLegalEntityId = @case.ApprenticeshipLegalEntityId,
+                    Status = @case.CaseStatus,
+                    CaseStatusLastUpdatedDate = @case.CaseStatusLastUpdatedDate
+                });
+        }
+
```

Comment:
Wouldn't it be better to use the domain event approach we use when the case is Completed?  That way there would be a single way of doing it.

Reply:
Is that something that could be picked up as part of the planned refactoring?